compiler/rustc_incremental/src/persist/fs.rs

```diff
@@ -657,62 +658,63 @@ pub fn garbage_collect_session_directories(sess: &Session) -> io::Result<()> {
             // This is something we don't know, leave it alone
         }
     }
+    session_directories.sort();
 
     // Now map from lock files to session directories
-    let lock_file_to_session_dir: FxHashMap<String, Option<String>> = lock_files
-        .into_iter()
-        .map(|lock_file_name| {
+    let lock_file_to_session_dir: UnordMap<String, Option<String>> =
+        UnordMap::from(lock_files.into_items().map(|lock_file_name| {
```

Comment:
can this be `.into()` instead, given that you have an explicit type annotation on the let binding?